.eslintrc.json

```diff
@@ -51,5 +51,13 @@
         "extensions": [".jsx", ".js", ".tsx", ".ts", ".scss", ".css", ".png", ".svg"]
       }
     }
-  }
+  },
+  "overrides": [
+    {
+      "files": ["*.ts", "*.tsx"],
+      "rules": {
+          "no-undef": "off"
+      }
+    }
+  ]
```

Comment:
am curious what overriding `no-undef` would do :o is it risky to disable the rule for all typescript files?

Reply:
Thanks for checking in about this. 

I ran into an issue with eslint when using `JSX` type from React. This is because `eslint` does not check globally defined variables (unlike Typescript) when checking for definitions, so it throws a no-defined error. I disabled the `eslint` check based on the [recommendations here](https://github.com/typescript-eslint/typescript-eslint/blob/main/docs/linting/TROUBLESHOOTING.md#i-get-errors-from-the-no-undef-rule-about-global-variables-not-being-defined-even-though-there-are-no-typescript-errors), quoted below:

> We strongly recommend that you do not use the no-undef lint rule on TypeScript projects. The checks it provides are already provided by TypeScript without the need for configuration - TypeScript just does this significantly better.

Since TypeScript is likely to provide more comprehensive and accurate checks on no-def, I think it'd be safe to disable eslint no-def checks. I'll also add the link inline for future reference. Thanks for clarifying!

Reply:
[ee5aac2](https://github.com/opengovsg/GoGovSG/pull/2040/commits/ee5aac22d6e49ee2ff215a9efe309ae5724543ad)